src/components/tree-item/tree-item.tsx

```diff
@@ -345,7 +345,8 @@ export class TreeItem implements ConditionalSlotComponent {
 
         // When focus is on a open node, moves focus to the first child node.
         if (this.hasChildren && this.expanded) {
-          this.el.querySelector("calcite-tree-item").focus();
+          this.el.querySelector("calcite-tree-item")?.focus();
+          e.stopPropagation();
```

Comment:
I see that this is an existing pattern in this implementation, but unless it's an internal event, let's avoid stopping propagation of events. It breaks expectations and could lead to other bugs (e.g., end-user expects to handle key event, but doesn't receive it, so they resort to listening on the capture phase, etc...).

Reply:
I'll try and see if there is a better way to solve this in the coming iterations on this bug.